spec/rubocop/cop/style/empty_literal_spec.rb

```diff
@@ -2,37 +2,26 @@
 
 RSpec.describe RuboCop::Cop::Style::EmptyLiteral, :config do
   describe 'Empty Array' do
-    it 'registers an offense for Array.new()' do
-      expect_offense(<<~RUBY)
-        test = Array.new()
-               ^^^^^^^^^^^ Use array literal `[]` instead of `Array.new`.
-      RUBY
-
-      expect_correction(<<~RUBY)
-        test = []
-      RUBY
-    end
-
-    it 'registers an offense for Array.new' do
-      expect_offense(<<~RUBY)
-        test = Array.new
-               ^^^^^^^^^ Use array literal `[]` instead of `Array.new`.
-      RUBY
+    shared_examples 'registers_and_corrects' do |initializer:|
+      it "registers an offense for #{initializer}" do
+        expect_offense(<<~RUBY)
+          test = #{initializer}
+                 #{'^' * initializer.length} Use array literal `[]` instead of `#{initializer}`.
+        RUBY
 
-      expect_correction(<<~RUBY)
-        test = []
-      RUBY
+        expect_correction(<<~RUBY)
+          test = []
+        RUBY
+      end
     end
 
-    it 'registers an offense for ::Array.new' do
-      expect_offense(<<~RUBY)
-        test = ::Array.new
-               ^^^^^^^^^^^ Use array literal `[]` instead of `Array.new`.
-      RUBY
-
-      expect_correction(<<~RUBY)
-        test = []
-      RUBY
+    context 'initializer resulting in an empty array literal' do
```

Comment:
combined existing and new cases into shared examples for readability

Reply:
Thank you for being considerate. However, as I mentioned in the comment below, `shared_examples` do not always improve readability. Can you respect the existing spec descriptions?

Reply:
I think in that particular case the shared examples are OK, given that we're essentially testing 6 times for the same thing.